sklearn/model_selection/tests/test_validation.py

```diff
@@ -226,6 +227,17 @@ def get_params(self, deep=False):
         return {'a': self.a, 'allow_nd': self.allow_nd}
 
 
+def test_array_function_not_called():
+    X = np.array([[1, 1], [1, 2], [1, 3], [1, 4],
+                  [2, 1], [2, 2], [2, 3], [2, 4],
+                  [3, 1], [3, 2], [3, 3], [3, 4]])
+    X = _NotAnArray(X)
+    y = _NotAnArray([1, 1, 1, 1, 2, 2, 2, 2, 1, 1, 2, 2])
+    estimator = LogisticRegression()
+    grid = GridSearchCV(estimator, param_grid={'C': [1, 10]})
+    cross_validate(grid, X, y, n_jobs=2)
```

Comment:
I do not think we need nested cross validation to check that `cross_validate` itself does not use `__array_function__`:

```suggestion
    cross_validate(estimator, X, y, cv=2)
```

Also `cv=2` to speed the test up and leaving `n_jobs=None` as default.